File: src/Website/Models/Database/Video.cs
using System;
using System.Collections.Generic;
using Website.Models.Validation.SubmitEpisode;

namespace Website.Models.Database
{
    public class Video
    {
        public string Id { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public DateTime Published { get; set; }
        public TimeSpan? Duration { get; set; }
        public List<Thumbnail> Thumbnails { get; set; } = new List<Thumbnail>();
        public bool RequiresUpdate { get; set; } = false;
        public int? Likes { get; set; } = null;
        public int? Dislikes { get; set; } = null;
        public int? ViewCount { get; set; } = null;
        public int? FavouriteCount { get; set; } = null;
        public int? CommentCount { get; set; } = null;
        public List<SubmittedEpisode> Submissions { get; set; } = new List<SubmittedEpisode>();
        public List<string> Tags { get; set; } = new List<string>();
        public bool Is3D { get; set; } = false;
        public bool IsHD { get; set; } = false;
        public bool HasCaption { get; set; } = false;
    }
}

File: src/Website/wwwroot/js/src/interfaces/video.ts
import { Thumbnail } from "./thumbnail";

export interface Video {
    id: string,
    title: string,
    published: string,
    duration: number,
    thumbnails: Array<Thumbnail>,
    requires_update: boolean,
    likes: number | null,
    dislikes: number | null,
    view_count: number | null,
    favorite_count: number | null,
    comment_count: number | null,
    submissions: null,
    tags: Array<string>,
    is_3d: boolean,
    is_hd: boolean,
    cc: boolean,
    submission_count: number,
    ratio: number | null
}